Tools/clinic/clinic.py

```diff
@@ -2272,44 +2280,61 @@ def compute_checksum(input, length=None):
 
 
 class PythonParser:
-    def __init__(self, clinic):
+    def __init__(self, clinic: Clinic) -> None:
         pass
 
-    def parse(self, block):
+    def parse(self, block: Block) -> None:
         s = io.StringIO()
         with OverrideStdioWith(s):
             exec(block.input)
         block.output = s.getvalue()
 
 
+ModuleDict = dict[str, "Module"]
+
 class Module:
-    def __init__(self, name, module=None):
+    def __init__(
+            self,
+            name: str,
+            module = None
+    ) -> None:
         self.name = name
         self.module = self.parent = module
 
-        self.modules = collections.OrderedDict()
-        self.classes = collections.OrderedDict()
-        self.functions = []
+        self.modules: ModuleDict = collections.OrderedDict()
+        self.classes: ClassDict = collections.OrderedDict()
```

Comment:
Since we're touching these lines anyway: do these need to be `OrderedDict`s anymore? Regular `dict`s are guaranteed to be ordered on 3.7+ (and clinic now requires 3.10+), and regular dicts are faster than `OrderedDict`s. `OrderedDict`s are mainly there for backwards compatibility these days (they have a few fancy methods that dicts don't have, but I really doubt we need any of them in `clinic.py`).

Reply:
Could leave this to another PR, though

Reply:
Yes, regular dicts should be fine post 3.7.